lib/src/kbasesearchengine/main/NarrativeInfoDecorator.java

```diff
@@ -139,6 +139,7 @@ public GetObjectsOutput getObjects(final GetObjectsInput params, final String us
         final Set<String> userNames = new HashSet<>();
         final Iterator<ObjectData> iter = objects.iterator();
         final Set<Long> seenWorkspaces = new HashSet<>();
```

Comment:
I see you set seenWorkspaces and deletedWorkspaces here here and then you check them a few lines later. Where do they get populated?a